core/src/main/java/org/arquillian/smart/testing/scm/git/GitChangeResolver.java

```diff
@@ -88,6 +89,26 @@ public boolean isApplicable() {
         return true;
     }
 
+    private Set<Change> retrieveCommitsChanges() {
+        try (ObjectReader reader = git.getRepository().newObjectReader()) {
+            final ObjectId oldHead = git.getRepository().resolve(previous + ENSURE_TREE);
+            final ObjectId newHead = git.getRepository().resolve(head + ENSURE_TREE);
+
+            final CanonicalTreeParser oldTree = new CanonicalTreeParser();
+            oldTree.reset(reader, oldHead);
+            final CanonicalTreeParser newTree = new CanonicalTreeParser();
+            newTree.reset(reader, newHead);
+
+            final List<DiffEntry> commitDiffs = git.diff()
+                    .setNewTree(newTree)
+                    .setOldTree(oldTree)
+                .call();
+            return transformToChangeSet(findRenames(commitDiffs), repoRoot);
```

Comment:
Aren't we just transforming the renames and not the other changes like add or modify?

Reply:
No, we transform all. `findRenames` will merge `ADD` and `DELETE` entries which are in fact `RENAME`s.

Reply:
Would be more understandable with different name? What would you suggest?

Reply:
Yes I think @dipak-pawar was confused as well. But I am not sure about another name, maybe `mergeRenames`?

Reply:
I was thinking the same, but it actually merges adds and deletes to renames based on similarity score.

Reply:
Then maybe add a comment saying this?

Reply:
That's exactly what I'm doing right now